### PR TITLE
Added role option in Snowflake doc

### PR DESCRIPTION
**Pull request by ishachopra0401**
## What do these changes do?

> Describe the changes you are proposing in this PR.

## Type of documentation

- [ ] New documentation
- [ ] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/sources/reverse-etl/snowflake.mdx [hunk 1]
```diff
@@ -157,6 +157,7 @@ If you have chosen the <strong>Model</strong> option, skip the next steps and re
     - **Database**: Enter the name of the database in which your data resides.
     - **Warehouse**: Specify the name of your data warehouse.
     - **User**: Enter the name of the user that has the required read/write access to the above database.
+    - **Role**: Enter the role of the user associated with the above account.
```
Comment (Vikas26021999) [inline docs/sources/reverse-etl/snowflake.mdx:160]: `Enter the role to be used with the above account` 
A single user can have multiple roles associated with them

Comment (Vikas26021999) [reply docs/sources/reverse-etl/snowflake.mdx:160]: We want to convey to the user that they can use any other role here, apart from default role that they have already set during the onboarding

**Review by Vikas26021999 — APPROVED** [review]
LGTM

File: docs/sources/reverse-etl/snowflake.mdx [hunk 2]
```diff
@@ -40,6 +40,10 @@ SHOW ROLES;
 CREATE USER RUDDER PASSWORD = '<strong_unique_password>' DEFAULT_ROLE = 'RUDDER_ROLE';
 ```
 
+<div class="infoBlock">
+You can assign some other role to this user later on in the RudderStack dashboard settings as specified in the <Link to="#configuring-the-connection-credentials">Configuring the connection credentials</Link> section.
```
Comment (ameypv-rudder) [inline docs/sources/reverse-etl/snowflake.mdx:44]:
```suggestion
You can assign some other role to this user in the RudderStack dashboard settings later, as specified in the <Link to="#configuring-the-connection-credentials">Configuring the connection credentials</Link> section.
```

File: docs/sources/reverse-etl/snowflake.mdx [hunk 3]
```diff
@@ -157,6 +161,7 @@ If you have chosen the <strong>Model</strong> option, skip the next steps and re
     - **Database**: Enter the name of the database in which your data resides.
     - **Warehouse**: Specify the name of your data warehouse.
     - **User**: Enter the name of the user that has the required read/write access to the above database.
+    - **Role**: Enter a role you want to assign to the above user. You can use this role for syncing data apart from the default role (`RUDDER_ROLE`) assigned in the <Link to="#step-1-creating-a-new-role-and-user-in-snowflake">Creating a new role and user</Link> section.
```
Comment (ameypv-rudder) [inline docs/sources/reverse-etl/snowflake.mdx:164]:
```suggestion
    - **Role**: Enter a role you want to assign to the above user. For syncing the data, you can use this role apart from the default role (`RUDDER_ROLE`) assigned in the <Link to="#step-1-creating-a-new-role-and-user-in-snowflake">Creating a new role and user</Link> section.
```

Comment (ameypv-rudder) [reply docs/sources/reverse-etl/snowflake.mdx:164]: For better readability and continuity - lmk if it makes sense.

**Review by ameypv-rudder — APPROVED** [review]

**Pull request merged**